main/src/library/Channel.flix

```diff
@@ -101,6 +101,14 @@ namespace Channel {
         a_id <=> b_id
     }
 
+    ///
+    /// Spawn the function `f` and send its value on the channel returned by this function.
+    ///
+    def spawnWithChannel(f: Unit -> a & ef): Channel[a] & Impure =
```

Comment:
This should not be impure, but have effect `ef` (with a cast).

Reply:
Also we need to move it after new channel creation. But it seems that this file will be split into two anyway.

Reply:
Hmm. I guess it should be impure / doesnt really matter, because if you use the chan then you will be impure anyway.

Reply:
This suggests that it would be better for you to allocate the channel yourself.

Reply:
I think it should be Impure (as it is now) since it isn't guaranteed that the thread spawned is closed upon return. Maybe this use case would be better supported by a very simple Future library that just has `def compute(f: Unit -> a): Future[a]` and `def getValue(f: Future[a]): a` that is blocking (i ignore effects here).